bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/SecurityCapability.java

```diff
@@ -0,0 +1,180 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.handler.capability;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.netatmo.internal.api.NetatmoException;
+import org.openhab.binding.netatmo.internal.api.SecurityApi;
+import org.openhab.binding.netatmo.internal.api.data.NetatmoConstants.FloodLightMode;
+import org.openhab.binding.netatmo.internal.api.dto.HomeData;
+import org.openhab.binding.netatmo.internal.api.dto.HomeDataModule;
+import org.openhab.binding.netatmo.internal.api.dto.HomeDataPerson;
+import org.openhab.binding.netatmo.internal.api.dto.HomeEvent;
+import org.openhab.binding.netatmo.internal.api.dto.HomeStatusModule;
+import org.openhab.binding.netatmo.internal.api.dto.HomeStatusPerson;
+import org.openhab.binding.netatmo.internal.api.dto.NAHomeStatus.HomeStatus;
+import org.openhab.binding.netatmo.internal.api.dto.NAObject;
+import org.openhab.binding.netatmo.internal.deserialization.NAObjectMap;
+import org.openhab.binding.netatmo.internal.handler.ApiBridgeHandler;
+import org.openhab.binding.netatmo.internal.handler.CommonInterface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SecurityCapability} is the base class for handler able to handle security features
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+class SecurityCapability extends RestCapability<SecurityApi> {
+    private final Logger logger = LoggerFactory.getLogger(SecurityCapability.class);
+
+    SecurityCapability(CommonInterface handler) {
+        super(handler);
+        ApiBridgeHandler bridgeApi = handler.getRootBridge();
+        if (bridgeApi != null) {
+            api = Optional.ofNullable(bridgeApi.getRestManager(SecurityApi.class));
+        }
+    }
+
+    @Override
+    protected void updateHomeData(HomeData homeData) {
+        NAObjectMap<HomeDataPerson> persons = homeData.getPersons();
+        NAObjectMap<HomeDataModule> cameras = homeData.getModules();
+        handler.getActiveChildren().forEach(handler -> {
+            HomeDataPerson dataPerson = persons.get(handler.getId());
+            if (dataPerson != null) {
+                handler.setNewData(dataPerson);
+            }
+            HomeDataModule data = cameras.get(handler.getId());
+            if (data != null) {
+                handler.setNewData(data);
+            }
+        });
+    }
+
+    @Override
+    protected void updateHomeStatus(HomeStatus homeStatus) {
+        NAObjectMap<HomeStatusPerson> persons = homeStatus.getPersons();
+        NAObjectMap<HomeStatusModule> cameras = homeStatus.getModules();
+        for (CommonInterface handler : handler.getActiveChildren()) {
```

Comment:
Is there a good reason to use for in this method and foreach in method updateHomeData ?
What is this reason ?

Reply:
Obviously not at all. Changed.